scipp-imaging-2D.ipynb

```diff
@@ -224,218 +162,516 @@
+    "        y = neighbour_tof[np.argmax(median_decision)]\n",
+    "\n",
+    "        # TODO ws.setE. Uncertanties not delt with\n",
+    "        filtered_var[\"spectrum\", spec_num].values = y\n",
+    "\n",
+    "    return filtered_var\n",
+    "    \n",
+    "\n",
+    "for k in wavelength.keys():\n",
+    "    if k == \"reference\":  # This can be removed once #1176 is resolved\n",
+    "        continue\n",
+    "\n",
+    "    wavelength[k] = median_filter(dataset=wavelength, variable_key=k)\n",
+    "    # TODO export_to_tiff_stack for each"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   }
   },
   {
    "cell_type": "code",
    "execution_count": null,
    "metadata": {
-    "scrolled": true
+    "pycharm": {
+     "name": "#%%\n"
+    }
    },
    "outputs": [],
    "source": [
-    "# Adds the component info needed for converting units\n",
-    "sc.compat.mantid.load_component_info(stitched, instrument_file)\n",
-    "stitched"
+    "# As each detector has a different relative position to the sample we\n",
+    "# need to rebin to a common set of wavelength bins for all\n",
+    "wav_x_vals = wavelength[\"sample\"].coords[\"wavelength\"].values\n",
+    "start_wave = np.max(wav_x_vals[:, 0])\n",
+    "stop_wave = np.min(wav_x_vals[:, -1])\n",
+    "bin_width_wave = np.mean(wav_x_vals[:, 1:]-wav_x_vals[:, :-1])\n",
+    "\n",
+    "wavelength_rebin_params = sc.Variable([\"wavelength\"], values=np.arange(start_wave, stop_wave, bin_width_wave,\n",
+    "                                                           dtype=np.float64))\n",
+    "\n",
+    "rebinned_wavelength = sc.rebin(wavelength, \"wavelength\", wavelength_rebin_params)\n",
+    "\n",
+    "# if True, the whole initial state is integrated, if False initial state is also grouped\n",
+    "integrating_of_sample_in_inital_state = True\n",
+    "if integrating_of_sample_in_inital_state:\n",
+    "    group_size = (324.0  / nx_target) * (324.0 / ny_target)\n",
+    "    factor = group_size\n",
+    "    rebinned_wavelength[\"sample\"] *= factor"
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
    "metadata": {
-    "scrolled": true
+    "pycharm": {
+     "name": "#%%\n"
+    }
    },
    "outputs": [],
    "source": [
-    "# Note: this cell is currently a workaround until \n",
-    "# https://github.com/scipp/scipp/issues/823 is done.\n",
-    "# Then it should be possible to group the positions label directly.\n",
-    "\n",
-    "# makes the position a DataConstProxy otherwise groupby won't take it\n",
-    "stitched[\"position\"] = stitched.labels[\"position\"]\n",
-    "dm1d = sc.groupby(stitched[\"position\"], \"detector_mapping\", Dim.Row)\n",
-    "position = dm1d.mean(Dim.Spectrum)\n",
+    "def weighted_sum(variable : sc.Variable, dim_to_sum):\n",
+    "    variable = variable.copy()\n",
```

Comment:
Why copy?

Reply:
I originally though sum worked in place so took a copy to be safe